neuro_san/service/interfaces/agent_server.py

```diff
@@ -0,0 +1,26 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+
+class AgentServer:
+    """
+    Interface for an AgentServer, regardless of transport mechanism
+    """
```

Comment:
Interface extracted from what is now called GrpcAgentServer so that the HttpSideCar can use the object via this interface instead of having a reach-around handle to something explicitly grpc.